### DISCUSSION TITLE
Use store not inside component.

### DISCUSSION THREAD

**Discussion by vvitto** [jonbnewman/mobx-store-provider#12] [Q&A] [answered]
Hi, first of all thank you for a cool library. I have a case when I need to call store action not inside of react component, if I will try to call useStore hook not inside react component it will trigger react error. Maybe you can suggest some way how to handle it. Maybe is it a good point to export 'stores' variable? Or just add a new helper function 'getStore' to incapsulate some logic with working with internal 'stores' variable?

**Comment by jonbnewman** [comment]
> Hi, first of all thank you for a cool library.

Thanks!

> I have a case when I need to call store action not inside of react component, if I will try to call useStore hook not inside react component it will trigger react error.

You are correct that this will cause an error currently because of the [Only Call Hooks from React Functions](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-from-react-functions) rule.

...and yes I believe this is a valid use case.

Also I think your proposed `getStore` works well as that would mirror `useStore` and its API:

```javascript
import { getStore } from 'mobx-store-provider';
import AppStore from 'stores/AppStore';

function someCallback() {
  const appStore = getStore(AppStore)
  // ...
}
```

This would not directly expose the internal `stores` map however it would allow you to get access to all of your models just the same. Which is what is needed here.

I will see if I can get this added in the next day or two...thanks for the suggestion!

**Comment by jonbnewman** [comment, marked as answer]
@vvitto ...just released `v2.1` that adds `getStore`.

http://mobx-store-provider.overfoc.us/api/getStore

Should provide what you needed, let me know if you have any problems.
